src/lib/ui/AccessCodeForm/AccessCodeForm.tsx

```diff
@@ -296,6 +319,38 @@ function getAccessCodeTimezone(
   return timezone
 }
 
+function getCodeLengthRequirement(device: CommonDevice): string | null {
+  if (!isLockDevice(device)) {
+    return null
+  }
+
+  const codeLengths = device.properties.supported_code_lengths
+  if (codeLengths == null) {
+    return null
+  }
+
+  if (codeLengths.length === 1) {
+    return `${codeLengths[0]}`
+  }
+
+  if (isSequential(codeLengths.join(''))) {
+    return `${codeLengths[0]}-${codeLengths[codeLengths.length - 1]}`
+  }
+
+  return codeLengths.join(', ')
+}
+
+function isSequential(numbers: string): boolean {
+  // 0 - 99 in a string
+  // 0123456789101112...99
+  const sequentialNumbers = Array.from(
```

Comment:
Probably not an real performance issue at this size, but better perhaps to initialize this array outside the function so it's only created once and not per-render.

Reply:
Updated in https://github.com/seamapi/react/pull/475/commits/e4267e8b1b0a92269f262a778e337e7a273e60f5